lib/_http_outgoing.js

```diff
@@ -575,6 +581,10 @@ OutgoingMessage.prototype.setHeader = function setHeader(name, value) {
   validateHeaderName(name);
   validateHeaderValue(name, value);
 
+  if (name === 'content-length') {
+    this._contentLength = NumberParseInt(value);
+  }
```

Comment:
Why? Isn’t store header enough?

Reply:
Oh sorry, shouldn't be there, my mistake

Reply:
I checked the code again and store header is not enough, the Content-length is not set when setHeader is used... I think it happens because it never goes throw processHeader

Reply:
Putting inside the validation function is wrong though.

Reply:
It seemed right sorry, I'll create a new function for the validating the content length